File: backend/InventoryApp.Api/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace InventoryApp.Api.Models;

public class Product
{
    public int Id { get; set; }

    [Required]
    [MaxLength(200)]
    public string Name { get; set; } = null!;

    [Required]
    [MaxLength(2000)]
    public string Description { get; set; } = null!;

    [Required]
    public decimal Price { get; set; }

    [Required]
    [MaxLength(50)]
    public string SKU { get; set; } = null!;

    public DateTime CreatedAt { get; set; }

    [ConcurrencyCheck]
    public DateTime UpdatedAt { get; set; }

    // Navigation property
    public Inventory? Inventory { get; set; }
} 

File: frontend/src/types/Product.ts
export interface Product {
  id: number;
  name: string;
  description?: string;
  price: number;
  stockQuantity: number;
  sku?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateProductDto {
  name: string;
  description?: string;
  price: number;
  stockQuantity: number;
  sku?: string;
}

export interface UpdateProductDto {
  name: string;
  description?: string;
  price: number;
  sku?: string;
}

export interface UpdateStockDto {
  quantity: number;
} 